File: backend/app/routers/fleet.py
from fastapi import APIRouter, Query

from app.mock_data import MOCK_EVENTS, MOCK_FAIRNESS, compute_fleet_summary

router = APIRouter(tags=["fleet"])


@router.get("/health")
async def health():
    return {"status": "ok"}


@router.get("/fleet/summary")
async def fleet_summary():
    # TODO: replace with an aggregation query against MongoDB's
    # charging_events collection once real data is loaded.
    return compute_fleet_summary()


@router.get("/charging-events")
async def charging_events(
    driver_id: str | None = Query(default=None, alias="driverId"),
    vehicle_id: str | None = Query(default=None, alias="vehicleId"),
):
    # TODO: replace with a MongoDB find() with optional filters.
    events = MOCK_EVENTS
    if driver_id:
        events = [e for e in events if e["driverId"] == driver_id]
    if vehicle_id:
        events = [e for e in events if e["vehicleId"] == vehicle_id]
    return events


@router.get("/fairness")
async def fairness_metrics():
    # TODO: replace with a real fairness computation over charging_events,
    # grouped by driver, comparing home/workplace/public cost exposure.
    return MOCK_FAIRNESS

File: frontend/src/api/fleet.ts
import { apiClient } from "./client";
import type { ChargingEvent, FairnessMetric, FleetSummary } from "../types/domain";

export async function getFleetSummary(): Promise<FleetSummary> {
  const { data } = await apiClient.get<FleetSummary>("/fleet/summary");
  return data;
}

export async function getChargingEvents(params?: {
  driverId?: string;
  vehicleId?: string;
}): Promise<ChargingEvent[]> {
  const { data } = await apiClient.get<ChargingEvent[]>("/charging-events", { params });
  return data;
}

export async function getFairnessMetrics(): Promise<FairnessMetric[]> {
  const { data } = await apiClient.get<FairnessMetric[]>("/fairness");
  return data;
}

export async function getHealth(): Promise<{ status: string }> {
  const { data } = await apiClient.get<{ status: string }>("/health");
  return data;
}
